src/project/modals/signup.modal.js

```diff
@@ -0,0 +1,133 @@
+
+import { expect } from 'playwright/test';
+export class SignupModal {
+    constructor(page) {
+        this.page = page;
+        this.nameField = this.page.locator('#signupName');
+        this.lastNameField = this.page.locator('#signupLastName');
+        this.emailField = this.page.locator('#signupEmail');
+        this.passwordField = this.page.locator('#signupPassword');
+        this.repeatPasswordField = this.page.locator('#signupRepeatPassword');
+        this.registrationBtn = this.page.locator('.modal-footer .btn-primary');
+        this.invalidFeedback = this.page.locator('.invalid-feedback').first();
+    }
+
+    async registerNewCustomer(registrationData) {
+        await this.fillNameField(registrationData.name);
+        await this.fillLastNameField(registrationData.last_name);
+        await this.fillEmailField(registrationData.email);
+        await this.fillPasswordField(registrationData.password);
+        await this.fillRepeatPasswordField(registrationData.password);
+        await this.registrationBtn.click();
+    }
+
+    async checkNameField(registrationData) {
+        await this.verifyInvalidName('123', 'Name is invalid');
+        await this.verifyInvalidName('', 'Name required');
+        await this.verifyInvalidName('k', 'Name has to be from 2 to 20 characters long');
+        await this.resetNameField(registrationData.name);
+    }
+
+    async checkLastNameField(registrationData) {
+        await this.verifyInvalidLastName('123', 'Last name is invalid');
+        await this.verifyInvalidLastName('', 'Last name required');
+        await this.verifyInvalidLastName('k', 'Last name has to be from 2 to 20 characters long');
+        await this.resetLastNameField(registrationData.last_name);
```

Comment:
you don't have any control under this method because data is hardcoded

data should be placed on scenario level

not on business logic layer